grafana/grafana
# Image Renderer: Add support for SSL in plugin mode

Address issue where SSL is not supported for plugin mode of image renderer.

Added a very basic test and verified fix locally :)

Before

https://github.com/user-attachments/assets/c7f4fc77-efca-4698-ab5f-b4b1e7786bda


After

https://github.com/user-attachments/assets/d050e8d3-26ba-4d35-a915-377887892429


TODO

- [x] Test and verify fix locally

Guide for local testing (on macOS):

1. Run image renderer as a plugin in local Grafana
- Pull the https://github.com/grafana/grafana-image-renderer repo locally
- Update the `Makefile` in the repo and update the `ARCH=` https://github.com/grafana/grafana-image-renderer/blob/122374c85741e8c6fa12c63ecfeacbf8964b7db6/Makefile#L3 with `darwin-arm64-unknown`
- In terminal in root of image renderer repo, run `make package`
- Copy the resulting plugin in the `dist` directory `plugin-darwin-arm64-unknown` into your `plugins` directory for local Grafana
- Rename the copied plugin to be `grafana-image-renderer`
- Add `[plugins] pluginsAllowUnsigned=true` to the `custom.ini` file in your local Grafana codebase
```
[plugins]
pluginsAllowUnsigned=true
```
- Install image renderer plugin in local Grafana instance
- To get local Grafana to run image renderer in plugin mode vs http mode you should comment out / remove the `server_url` config under the `[rendering]` section
```
[rendering]
; server_url = http://localhost:8081/render
```

You now should be able to test the image renderer by opening a dashboard, clicking the triple dots on a panel, click "share" -> "share link" and then click the generate image button.

2. Run your local grafana with https
- See https://www.turbogeek.co.uk/grafana-how-to-configure-ssl-https-in-grafana/ for more detail. Be sure to update the [server] protocol to be `https` in the Grafana local repo `custom.ini` file
- For generating certs and allowing this locally on mac to work see https://stackoverflow.com/questions/62010077/set-up-https-on-localhost-for-macos-mac-os-catalina-10-15-2/73517777#73517777
- Copy the generated key + cert file into the root of grafana repo and update the [server] config items in the `custom.ini` file
- Once you have localhost working with https you can now test and reproduce the issue and verify this PR fixes the issue.
- On `main` branch with https localhost try and generate an image. This will result in a cert error due to image renderer trying to use http for localhost
- Now switch to this PR's branch and set the `callback_url` under the `[rendering]` section of `custom.ini` to be `https//localhost:3000/`
- Now try and generate an image... it should work!

Fixes https://github.com/grafana/grafana/issues/89856
Should address  https://github.com/grafana/support-escalations/issues/13823
